### PR TITLE
[GNA] Enable Zero Padding

### PR DESCRIPTION
**Pull request by kbruniec**
+ Fix CNN2D stride order
   + Enable pooling
   + Add tests including one based on the real life model structure.

### CODE REVIEW

**Comment by kbruniec** [comment]
I recommend to merge

File: inference-engine/src/gna_plugin/gna_graph_compiler.cpp [hunk 1]
```diff
@@ -623,10 +642,10 @@ void GNAGraphCompiler::finalizeConvolution2DPrimitive(InferenceEngine::CNNLayerP
     // TODO: Jira: 43659 - the issue also appears when after conv there's an eltwise or activation
     // For last layer or when next ones are only non functional, the data can be reordered when exporting scores
     // For other cases inserting permute is required if data are reordered
-    auto isNonFunctional = [](CNNLayerPtr l) {
-        return LayerInfo(l).isNonFunctional();
+    auto keepsLayout = [](CNNLayerPtr l) {
```
Comment (elilobanova) [inline inference-engine/src/gna_plugin/gna_graph_compiler.cpp:645]: I suppose pooling can decrease a tensor size, so num_rotate_output_rows calculated on the base of convolution outputs will be incorrect.
At the same time I'm not sure if filling of rotation parameters is actually needed here. Now it's handled in gna_plugin.cpp and covers cases when there are any other layers after convolution, so I removed this code from 1d convolution primitive handling. Is there some special reason to keep it for 2d convolution?

File: inference-engine/src/gna_plugin/gna_graph_compiler.hpp [hunk 2]
```diff
@@ -36,7 +36,8 @@ class GNAGraphCompiler {
     std::shared_ptr<GNAPluginNS::InputDesc> inputDesc;
     std::shared_ptr<GNAPluginNS::GNAFlags> gnaFlags;
     Policy policy;
-
+    //consider moving to configuration
+    const bool cnn2dInputPaddingSupported = true;
```
Comment (bsochack) [inline inference-engine/src/gna_plugin/gna_graph_compiler.hpp:40]: It should be in policy, see gna_plugin_policy.hpp

**Review by bsochack — COMMENTED** [review]
- missing tests
- this change is not only about zero padding. It also adds FP32 implementation

File: inference-engine/src/gna_plugin/gna_graph_compiler.hpp (hunk 2 above)
Comment (kbruniec) [reply inference-engine/src/gna_plugin/gna_graph_compiler.hpp]: Moved

File: inference-engine/src/gna_plugin/gna_graph_compiler.cpp (hunk 1 above)
Comment (kbruniec) [reply inference-engine/src/gna_plugin/gna_graph_compiler.cpp]: You were right it is not needed. Fixed.

**Comment by kbruniec** [comment]
@elilobanova, @bsochack, As you reviewed this PR and your issues were addressed could you take a look and possibly approve?

**Review by elilobanova — APPROVED** [review]

**Pull request merged**